src/Symfony/Component/Serializer/Serializer.php

```diff
@@ -297,6 +306,11 @@ private function getDenormalizer($data, string $class, ?string $format, array $c
         return null;
     }
 
+    private function scalarDenormalization($type)
+    {
+        return \in_array($type, ['int', 'integer', 'bool', 'boolean', 'float', 'string'], true);
```

Comment:
This code is performance sensitive (it will be called recursively on the hot path) so I suggest to:

* move the list of type in a `const`
* transform this array in map (`const TYPES = ['int' => true, 'integer' => true/*...*/]`
* call `isset(self::TYPES[$type])` directly to remove an extra function call

Reply:
And isn't `double` missing?

Reply:
> And isn't `double` missing?

I thought that `double` is outdated version of `float`, but `gettype()` returning `double` for float values, so I think that `double` should be presented in types map.